cluster/photon-controller/util.sh

```diff
@@ -0,0 +1,995 @@
+        "curl ${curl_creds} ${curl_output} ${curl_net} https://${KUBE_MASTER_IP}/healthz"
+}
+
+#
+# Wait for the Kubernetes healthz API to be responsive on each node
+#
+function wait-node-apis {
+    local curl_output="--fail --output /dev/null --silent"
+    local curl_net="--max-time 1"
+
+    for (( i=0; i<${#NODE_NAMES[@]}; i++)); do
+        try-until-success "Waiting for Kubernetes API on ${NODE_NAMES[$i]}..." \
+            "curl ${curl_output} ${curl_net} http://${KUBE_NODE_IP_ADDRESSES[$i]}:10250/healthz"
+    done
+}
+
+#
+# Configure the nodes so the pods can communicate
+# Each node will have a bridge named cbr0 for the NODE_IP_RANGES
+# defined in config-default.sh. This finds the IP subnet (assigned
+# by Kubernetes) to nodes and configures routes so they can communicate
+#
+# Also configure the master to be able to talk to the nodes. This is
+# useful so that you can get to the UI from the master.
+#
+function setup-pod-routes {
+    local node
+
+    KUBE_NODE_BRIDGE_NETWORK=()
+    for (( node=0; node<${#NODE_NAMES[@]}; node++)); do
+
+        # This happens in two steps (wait for an address, wait for a non 172.x.x.x address)
+        # because it's both simpler and more clear what's happening.
+        try-until-success-ssh ${KUBE_NODE_IP_ADDRESSES[$node]} \
+            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]} to have an address"  \
+            'sudo ifconfig cbr0  | grep -oP "inet addr:\K\S+"'
+
+        try-until-success-ssh ${KUBE_NODE_IP_ADDRESSES[$node]} \
+            "Waiting for cbr0 bridge on ${NODE_NAMES[$node]} to have correct address"  \
+            'sudo ifconfig cbr0  | grep -oP "inet addr:\K\S+" | grep -v  "^172."'
+
+        run-ssh-cmd ${KUBE_NODE_IP_ADDRESSES[$node]} 'sudo ip route show | grep -E "dev cbr0" | cut -d " " -f1'
+        KUBE_NODE_BRIDGE_NETWORK+=($_OUTPUT)
+        kube::log::status "cbr0 on ${NODE_NAMES[$node]} is ${_OUTPUT}"
+    done
+
+    local i
+    local j
+    for (( i=0; i<${#NODE_NAMES[@]}; i++)); do
+        kube::log::status "Configuring pod routes on ${NODE_NAMES[$i]}..."
+        run-ssh-cmd ${KUBE_MASTER_IP} "sudo route add -net ${KUBE_NODE_BRIDGE_NETWORK[$i]} gw ${KUBE_NODE_IP_ADDRESSES[$i]}"
+        for (( j=0; j<${#NODE_NAMES[@]}; j++)); do
+            if [[ $i != $j ]]; then
+                run-ssh-cmd ${KUBE_NODE_IP_ADDRESSES[$i]} "sudo route add -net ${KUBE_NODE_BRIDGE_NETWORK[$j]} gw ${KUBE_NODE_IP_ADDRESSES[$j]}"
+            fi
+        done
+    done
+}
+
+#
+# Copy the certificate/key from the Kubernetes master
+# These are used to create the kubeconfig file, which allows
+# users to use kubectl easily
+#
+function copy-kube-certs {
+    local cert="kubecfg.crt"
+    local key="kubecfg.key"
+    local ca="ca.crt"
+    local cert_dir="/srv/kubernetes"
+
+    kube::log::status "Copying credentials from ${KUBE_MASTER}"
+
+    KUBE_CERT="${KUBE_TEMP}/${cert}"
+    KUBE_KEY="${KUBE_TEMP}/${key}"
+    CA_CERT="${KUBE_TEMP}/${ca}"
```

Comment:
I'll add a comment here to make this clear. Unfortunately, this is required by common.sh's create-kube-config function.